### PR TITLE
CI: Add macOS M1 CI

### LINKED ISSUE #47443: CI/PERF: Windows builds fail due to potential MemoryErrors
Debugging attempt: https://github.com/pandas-dev/pandas/pull/47318

First PR exhibiting these failures: https://github.com/pandas-dev/pandas/pull/47299

_May_ point to a potentially memory leak in recent non-nano changes.

### PR DESCRIPTION
**Pull request by lithomas1**
- [ ] closes #47443 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -227,8 +228,7 @@ jobs:
       PANDAS_CI: 1
       PYTEST_TARGET: pandas
       PATTERN: "not slow and not db and not network and not single_cpu"
-      # GH 47443: PYTEST_WORKERS > 0 crashes Windows builds with memory related errors
-      PYTEST_WORKERS: ${{ matrix.os == 'macos-latest' && 'auto' || '0' }}
+      PYTEST_WORKERS: 'auto'
```
Comment (mroeschke) [inline .github/workflows/unit-tests.yml:231]: Great we don't need this anymore!

File: .github/workflows/wheels.yml [hunk 2]
```diff
@@ -137,11 +139,15 @@ jobs:
         shell: bash -el {0}
         run: echo "sdist_name=$(cd ./dist && ls -d */)" >> "$GITHUB_ENV"
 
+      - name: Install pipx
```
Comment (mroeschke) [inline .github/workflows/wheels.yml:142]: Could you link https://github.com/actions/runner-images/issues/9256?

**Review by WillAyd — APPROVED** [review]
lgtm ex @mroeschke comments

**Comment by lithomas1** [comment]
self-merging as this is green. 

As always, happy to address comments in a followup.

**Pull request merged**